### ISSUE TITLE
Issue with nested data custom validation. An error is thrown when the path is not explicitly passed.

### ISSUE THREAD

**Issue by unlocomqx** [ciscoheat/sveltekit-superforms#431]
- [x] Before posting an issue, read the [FAQ](https://superforms.rocks/faq) and search the previous issues.

**Description**
When passing an error message in a nested path without an explicit path, an error is thrown

<details>
<summary>Error</summary>

SchemaError: [options.2.label] No default value specified for field (can be undefined, but must be explicit)
    at Defaults_traverseAndReplace (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/sveltekit-superforms@2.13.1_@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2_xcpgw6wb3g5ifyop26ipqrd7xe/node_modules/sveltekit-superforms/dist/errors.js:221:23)
    at Errors_traverseAndReplace (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/sveltekit-superforms@2.13.1_@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2_xcpgw6wb3g5ifyop26ipqrd7xe/node_modules/sveltekit-superforms/dist/errors.js:182:13)
    at Data_traverse (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/sveltekit-superforms@2.13.1_@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2_xcpgw6wb3g5ifyop26ipqrd7xe/node_modules/sveltekit-superforms/dist/errors.js:170:9)
    at Module.replaceInvalidDefaults (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/sveltekit-superforms@2.13.1_@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2_xcpgw6wb3g5ifyop26ipqrd7xe/node_modules/sveltekit-superforms/dist/errors.js:228:16)
    at Module.superValidate (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/sveltekit-superforms@2.13.1_@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2_xcpgw6wb3g5ifyop26ipqrd7xe/node_modules/sveltekit-superforms/dist/superValidate.js:38:11)
    at async load (/Users/USER/code/tests/svelte/superforms-nested-validation/src/routes/+page.server.ts:27:18)
    at async Module.load_server_data (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2.17_vite@5.2.11__svelte@4.2.17_vite@5.2.11/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:61:17)
    at async eval (/Users/USER/code/tests/svelte/superforms-nested-validation/node_modules/.pnpm/@sveltejs+kit@2.5.9_@sveltejs+vite-plugin-svelte@3.1.0_svelte@4.2.17_vite@5.2.11__svelte@4.2.17_vite@5.2.11/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:140:13) {
  path: 'options.2.label'
}

</details>

[Repro](https://github.com/unlocomqx/superforms-nested-validation)

Specifying a path resolves the issue but the errors and the data will have inconsistent shapes

```javascript
const schema = z.object({
    id: z.string(),
    options: z.record(z.string(), z.object({
        label: z.string().refine(value => value.length > 0, {
            message: "Label is required",
            path: ['error']
        }),
    })),
})
```

**Comment by unlocomqx** [comment]
Possible cause. Nested props have `additionalProperties` instead of `properties`.

![CleanShot 2024-05-20 at 18 58 07@2x](https://github.com/ciscoheat/sveltekit-superforms/assets/793712/75f3ed7f-bae7-4f44-a0c4-a934fabdf66c)

**Comment by unlocomqx** [comment]
Possible fix in https://github.com/ciscoheat/sveltekit-superforms/blob/main/src/lib/jsonSchema/schemaInfo.ts#L60

```ts
const currentSchema: JSONSchema7 = (schema.additionalProperties ? schema.additionalProperties : schema) as JSONSchema7;
```
`schema.additionalProperties` seems to have the same shape as a json schema

**Comment by ciscoheat** [comment]
Should be fixed now with 2.14.0, can you confirm that it is?

**Comment by unlocomqx** [comment]
Yes! Thank you for this fix. It's perfect. Both the repro and my project worked perfectly after the update!

Cheers 👋
